src/main/scala/uk/gov/nationalarchives/dp/client/ValidateXmlAgainstXsd.scala

```diff
@@ -0,0 +1,42 @@
+package uk.gov.nationalarchives.dp.client
+
+import cats.Monad
+import cats.implicits.given
+
+import java.io.StringReader
+import javax.xml.XMLConstants
+import javax.xml.parsers.DocumentBuilderFactory
+import javax.xml.transform.dom.DOMSource
+import javax.xml.transform.stream.StreamSource
+import javax.xml.validation.SchemaFactory
+import javax.xml.validation.Validator
+import scala.xml.InputSource
+
+trait ValidateXmlAgainstXsd[F[_]](pathOfSchemaFile: String) {
+  def xmlStringIsValid(xmlStringToValidate: String): F[Boolean]
+}
+
+object ValidateXmlAgainstXsd {
+  val xipXsdSchemaV6 = "/XIP-V6.xsd"
```

Comment:
You don't need this variable here, it isn't being used. It can be moved to the test class

Reply:
I added it here for the purpose of external usage (e.g. opex generation lambdas). The thinking was to build up a list of Preservica schemas so that when you call the validator, you can just pass in the schema that you want to use. What do you think? Should I maybe turn it into an enum so that it's clearer?

Reply:
An enum might make sense actually. See what you think about my other comment and then maybe change this to:
```scala
enum Schemas(schemaFileAsString: String):
    case xipXsdSchemaV6 extends Schemas(getClass.getResourceAsStream("/xsd").readAllBytes().map(_.toChar).mkString)
```

Reply:
But you've converted the file contents into a string, what about this line

```
streamSource <- Monad[F].pure(new StreamSource(getClass.getResourceAsStream(pathOfSchemaFile)))
```
?

Should I leave off `.readAllBytes().map(_.toChar).mkString`?